shell/mixins/chart-product-redirect.js

```diff
@@ -0,0 +1,49 @@
+import { REPO_TYPE, REPO, CHART, VERSION } from '@shell/config/query-params';
+
+export default function(product, chartName, install = true) {
+  return {
+    data() {
+      return { redirectPending: true };
```

Comment:
I really don't like this, if you have suggestions to remove `redirectPending` I'd like to hear it.

This is in place because on the pages which have content to render switching from the middleware to the mixin would cause the content to flash before performing a redirect.

The reason for this is because the pages wouldn't render until middleware was fully executed. This version on the other hand doesn't block rendering implicitly so we needed a way to hide the content until all the redirect logic finished executing.

Reply:
> About the suggestions, could we not take out all this logic and use the state management to trigger an action and get the state for the loading?
Why is this an exception in our loading state scenario using fetch()?
Also considering the repetition of the markup, should we not have some kind of component with slots or a routing guard for this?

I think you're right. It's related to the issue you opened a few weeks ago where we want to refactor loading. I think pushing the loading logic into state management is the right thing to do. I'm considering doing that when removing $loading from our codebase.

There is a subtle different in this case because the original code didn't actually have a loading screen for these page components. The page would just stay blank until the middleware finished and it would either render the current page or redirect.

I am still debating if changing beforeCreate to fetch would be better for this.

Reply:
Then probably better to take a second look later for the loading. It's a bit weird that we keep a blank page, but not doing UX at this round 😅 

I am personally indifferent between `fetch()` and `beforeCreate()`, plus in our code we have both, so I would rather stick to what you did, since it would be purely Vue functions at least.

Reply:
I don't have any direct recommendations for `redirectPending`, but if I'm reading this correctly, we might be able to consider implementing a per-route guard[^1] using `beforeEnter()` to keep the behavior similar to as it was before. It appears that access to the store in a route guard would be the primary blocker in accomplishing this, but our `useStore()` composable[^2] might be able to get the job done.  

This would require us to replace this mixin entirely and I'm not asking that we do this, but I can envision how it would remove the `redirectPending` requirement.

[^1]: https://v3.router.vuejs.org/guide/advanced/navigation-guards.html#per-route-guard

[^2]: https://github.com/rancher/dashboard/blob/master/shell/composables/useStore.ts

Reply:
The concept of having route guards in a dedicated section for routing would be optimal.

Reply:
Since we now have that dedicated section I went ahead and gave it a try moving this to beforeEnter(). As of right now beforeEnter() gets called before we load the schemas via the authenticated middleware.

Also unfortunately the mixin doesn't work anymore because of recent changes which prevents the component from ever rendering and thus never executed. This still works in master because middleware doesn't require the component to be rendered before it's executed.

I'm going to update this PR with the beforeEnter solution once the loadManagement and product setting is moved to navigation guards in master.